src/content/blog/2023_qrack_raid_0.md

```diff
@@ -0,0 +1,39 @@
+
+Believe it or not, for about $2,500, you can build your “dream” rig for gate-based quantum computer simulation up to 40 qubits and beyond (albeit significantly slower than a GPU cluster, but close to the processing throughput rate of a large CPU)! In this blog post, we’ll describe the construction of such a device and introduce you to how Unitary Fund’s Qrack simulator can use “novel” and approximate simulation techniques to push about 16 TB of high-speed swap disk from 40 qubits of state vector simulation into 54-qubit territory, potentially rivaling the 2019 Sycamore “quantum supremacy” experiments for fidelity and cost!
+
+IBM was likely the first to point out, after the 2019 Sycamore universal random circuit sampling experiments, that disk storage could be employed to increase effective RAM limits of quantum computer simulation far past what is attainable with GPU and “DIMMs” (RAM boards inserted into your computer motherboard). Roughly, the largest supercomputers in the world, even arrayed together, might only have DRAM capacity for about 49 or 50 qubits of state vector simulation, aside from other techniques like tensor networks or any approximate simulation techniques. (Just as with your personal computer hardware, adding 1 qubit _doubles_ the necessary memory footprint, and the capacity of the largest supercomputers in the world, today, does not double in negligible time, effort, or cost!) However, disk storage can be _nearly_ as good as “DIMMs,” but drastically cheaper and far easier to scale to terabyte-range and past!
+
+The obvious problem is that data transfer to and from disk is _far_ slower than “DIMMs.” Ultimately, “swap disk” as (random-access) memory needs to somehow be made fast enough to keep up with the processing throughput capacity of whatever CPU(s) to which it’s attached. However, Unitary Fund and the Qrack simulator development team have demonstrated proof-of-concept for creating swap disk resources that are _nearly_ fast enough to keep up with a 64-hyper-thread AMD Epyc processor: this prototype machine uses “RAID 0” configuration on 8 (“consumer-grade”) NVMe solid-state drives, achieving 8 times the throughput of a single drive! By “striping” the (single) logical swap disk partition across 8 hardware drives, interspersing small segments of memory in a pattern of “stripes” that are likely to be accessed at once across physical drives, the overall transfer speed nets to the _sum_ of the transfer speeds of each respective disk, achieving speeds close to sufficient to keep the CPU barely fully utilized on large state vector simulations (particularly with OS-level swap disk compression enabled, such as by manually enabling “zswap” with “zstd” compressor in Linux)!
```

Comment:
Add link? Suggestions: arXiv report and or previous blog post on Qrack white paper.